### ISSUE TITLE
Missing device configuration: Eurotronic Comet Z-Wave

### ISSUE THREAD

**Issue by cadwizzard** [AlCalzone/node-zwave-js#5464]
### Checklist

- [X] It is not in the [configuration DB](https://devices.zwave-js.io/?search=3)

- [X] It was not [merged recently](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Apr+is%3Aclosed) or has a [pending PR](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Aopen+is%3Apr)

### Which device is missing?

Eurotronic Comet Z-wave

### Manufacturer ID

0x0148

### Product Type

0x0004

### Product ID

0x0003

### Firmware Version

11.26.4

### Is the device listed on the Z-Wave Alliance website?

https://products.z-wavealliance.org/products/1601

### Do you have a manual?

https://manual.zwave.eu/backend/make.php?lang=en&sku=EURECOMET&cert=

*The device is detected but manufacturer does not show in Home Assistant, looks like a product ID increment as the config DB has product 1,2 and not 3*
The parameters are enumerated in discovery, as below, but are not editable or named to edit once interview completes (attached)

2023-02-14T18:03:52.164Z CNTRLR « [Node 113] received information for parameter #1:
                                  parameter name:      LCD_invert
                                  value format:        UnsignedInteger
                                  value size:          1 bytes
                                  min value:           0
                                  max value:           1
                                  default value:       0
                                  is read-only:        false
                                  is advanced (UI):    false
                                  has bulk support:    true
                                  alters capabilities: false
![comet z no parameter edit](https://user-images.githubusercontent.com/7688705/218826642-99b8b8ba-fb8d-44db-bb23-cf864ca5e10a.png)
![comet z](https://user-images.githubusercontent.com/7688705/218826668-5aac19d8-b78d-468f-a812-80272793d203.png)

**Comment by AlCalzone** [comment]
@zwave-js-bot import config 1601 from zwa

**Comment by AlCalzone** [comment]
Hm, automatically importing the config was not helpful. The IDs are a bit different than what we currently have for the Comet Z, but that can happen. We also don't have any config params defined at the moment, so they should be added too.

**Comment by cadwizzard** [comment]
If it helps, this is a snippet from the interview log:

      "index": 0,
      "installerIcon": 4608,
      "userIcon": 4608,
      "status": 4,
      "ready": true,
      "isListening": false,
      "isRouting": true,
      "isSecure": false,
      "manufacturerId": 328,
      "productId": 3,
      "productType": 4,
      "firmwareVersion": "11.26.4",
      "zwavePlusVersion": 2

I Cross checked all the states (in Home Assistant) against a Eurotronic Spirit and they are almost identical parameter and feature wise, except some extra preset mode state ("unknown (0x10") which is new/different:

hvac_modes: off, heat
min_temp: 7
max_temp: 35
preset_modes: none, Energy heat, Full power, unknown (0x10)
current_temperature: 20
temperature: null
preset_mode: none
friendly_name: TRV 22
supported_features: 17

The device seems to work perfectly with exception to the manufacturer/model  not displaying and most importantly, the config parameters are not editable. inverting LCD etc (param#1) means I cant use these until the device is working properly as everything is upside down on vertically mounted radiators. Also they work best with the temp sensor increment set to .3 rather than .5.

**Comment by cadwizzard** [comment]
@AlCalzone Is there anything I can do to help get the config parameters added, so they can be edited in Home assistant?
I'm happy to help as I can't use these TRV's without the ability to alter some of the parameters (like invert display etc)

**Comment by AlCalzone** [comment]
Yeah, see the comment above: https://github.com/zwave-js/node-zwave-js/issues/5464#issuecomment-1430241560

(FYI you can freely set config parameters in Z-Wave JS UI, you need to know the number and size though)

**Comment by cadwizzard** [comment]
Edit from : Thanks, I'll try to get what you need. 
Update: The way to get the config files and use IDE editors and yarn  is way above my skill level and I dont think I have access to platforms to even run the tasks. There is no way i'll be able to compile anything I have any idea even looks correct.
If anyone else takes this onboard i'll happily pull data from the included device or test for them.
I don't use JS UI, just the supported non UI zwave-js, so I dont believe I have access to parameter edit either.

**Comment by kpine** [comment]
> I don't use JS UI, just the supported non UI zwave-js, so I dont believe I have access to parameter edit either.

https://community.home-assistant.io/t/how-to-set-parameter-6-on-homeseer-hs-ws200-switches/431134/2

**Comment by cadwizzard** [comment]
@kpine Thanks!
great workaround until the device gets proper support :)

**Comment by bronger** [comment]
If I set manually in zwave-js-ui the Parameter 8 for the temperature offset, I get a “Parameter 8 set failed” in return. Is this because the device is not yet fully supported? I would have thought that setting parameters manually always works.

zwave-js-ui: 8.21.1.103c5be
zwave-js: 11.5.3

![image](https://github.com/zwave-js/node-zwave-js/assets/2184938/43eaeaa9-02c3-4467-a648-aedfc601d430)

**Comment by bronger** [comment]
Sorry, this was a mistake on my side. While the command still seems to be ignored, I cannot reproduce the error message any more. Probably I mixed up devices in the UI.

**Comment by cadwizzard** [comment]
In case this info helps anyone who can make the config for this - I moved to JSUI as I couldn't get the parameters working otherwise .
![comet](https://github.com/zwave-js/node-zwave-js/assets/7688705/36e57560-805b-4e69-940e-aedd84720ad5)
(for some reason 'add code' isnt formatting the below like it should)
```
2023-09-17T17:36:10.413Z - INTERVIEW STARTED
2023-09-17T17:36:10.472Z - INTERVIEW STAGE COMPLETED
Arg 0:
ProtocolInfo
2023-09-17T17:36:11.772Z - INTERVIEW STAGE COMPLETED
Arg 0:
NodeInfo
2023-09-17T17:36:11.827Z - VALUE ADDED
Arg 0:
└─commandClassName: Manufacturer Specific
└─commandClass: 114
└─property: manufacturerId
└─endpoint: 0
└─newValue: 328
└─propertyName: manufacturerId
2023-09-17T17:36:11.830Z - VALUE ADDED
Arg 0:
└─commandClassName: Manufacturer Specific
└─commandClass: 114
└─property: productType
└─endpoint: 0
└─newValue: 4
└─propertyName: productType
2023-09-17T17:36:11.833Z - VALUE ADDED
Arg 0:
└─commandClassName: Manufacturer Specific
└─commandClass: 114
└─property: productId
└─endpoint: 0
└─newValue: 3
└─propertyName: productId
2023-09-17T17:36:11.955Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: libraryType
└─endpoint: 0
└─newValue: 3
└─propertyName: libraryType
2023-09-17T17:36:11.959Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: protocolVersion
└─endpoint: 0
└─newValue: 7.15
└─propertyName: protocolVersion
2023-09-17T17:36:11.962Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: firmwareVersions
└─endpoint: 0
└─newValue: 11.26, 1.4
└─propertyName: firmwareVersions
2023-09-17T17:36:11.965Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: hardwareVersion
└─endpoint: 0
└─newValue: 1
└─propertyName: hardwareVersion
2023-09-17T17:36:13.044Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: sdkVersion
└─endpoint: 0
└─newValue: 7.15.4
└─propertyName: sdkVersion
2023-09-17T17:36:13.050Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: applicationFrameworkAPIVersion
└─endpoint: 0
└─newValue: 10.15.4
└─propertyName: applicationFrameworkAPIVersion
2023-09-17T17:36:13.058Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: applicationFrameworkBuildNumber
└─endpoint: 0
└─newValue: 256
└─propertyName: applicationFrameworkBuildNumber
2023-09-17T17:36:13.062Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: hostInterfaceVersion
└─endpoint: 0
└─newValue: unused
└─propertyName: hostInterfaceVersion
2023-09-17T17:36:13.070Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: hostInterfaceBuildNumber
└─endpoint: 0
└─newValue: 0
└─propertyName: hostInterfaceBuildNumber
2023-09-17T17:36:13.075Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: zWaveProtocolVersion
└─endpoint: 0
└─newValue: 7.15.4
└─propertyName: zWaveProtocolVersion
2023-09-17T17:36:13.079Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: zWaveProtocolBuildNumber
└─endpoint: 0
└─newValue: 256
└─propertyName: zWaveProtocolBuildNumber
2023-09-17T17:36:13.084Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: applicationVersion
└─endpoint: 0
└─newValue: 11.26.4
└─propertyName: applicationVersion
2023-09-17T17:36:13.088Z - VALUE ADDED
Arg 0:
└─commandClassName: Version
└─commandClass: 134
└─property: applicationBuildNumber
└─endpoint: 0
└─newValue: 43707
└─propertyName: applicationBuildNumber
2023-09-17T17:36:13.217Z - VALUE ADDED
Arg 0:
└─commandClassName: Battery
└─commandClass: 128
└─property: level
└─endpoint: 0
└─newValue: 85
└─propertyName: level
2023-09-17T17:36:13.220Z - VALUE ADDED
Arg 0:
└─commandClassName: Battery
└─commandClass: 128
└─property: isLow
└─endpoint: 0
└─newValue: false
└─propertyName: isLow
2023-09-17T17:36:13.720Z - VALUE ADDED
Arg 0:
└─commandClassName: Thermostat Mode
└─commandClass: 64
└─property: mode
└─endpoint: 0
└─newValue: 1
└─propertyName: mode
2023-09-17T17:36:13.946Z - VALUE ADDED
Arg 0:
└─commandClassName: Thermostat Setpoint
└─commandClass: 67
└─property: setpoint
└─propertyKey: 1
└─endpoint: 0
└─newValue: 8.5
└─propertyName: setpoint
└─propertyKeyName: Heating
2023-09-17T17:36:14.009Z - VALUE ADDED
Arg 0:
└─commandClassName: Thermostat Setpoint
└─commandClass: 67
└─property: setpoint
└─propertyKey: 11
└─endpoint: 0
└─newValue: 16
└─propertyName: setpoint
└─propertyKeyName: Energy Save Heating
2023-09-17T17:36:14.098Z - VALUE ADDED
Arg 0:
└─commandClassName: Protection
└─commandClass: 117
└─property: local
└─endpoint: 0
└─newValue: 0
└─propertyName: local
2023-09-17T17:36:14.270Z - VALUE ADDED
Arg 0:
└─commandClassName: Multilevel Sensor
└─commandClass: 49
└─property: Air temperature
└─endpoint: 0
└─newValue: 24
└─propertyName: Air temperature
2023-09-17T17:36:15.411Z - VALUE ADDED
Arg 0:
└─commandClassName: Multilevel Switch
└─commandClass: 38
└─property: currentValue
└─endpoint: 0
└─newValue: 0
└─propertyName: currentValue
2023-09-17T17:36:16.862Z - VALUE ADDED
Arg 0:
└─commandClassName: Configuration
└─commandClass: 112
└─property: 1
└─endpoint: 0
└─newValue: 1
└─propertyName: LCD_invert
2023-09-17T17:36:16.936Z - VALUE ADDED
Arg 0:
└─commandClassName: Configuration
└─commandClass: 112
└─property: 2
└─endpoint: 0
└─newValue: 5
└─propertyName: LCD_timeout
2023-09-17T17:36:16.999Z - VALUE ADDED
Arg 0:
└─commandClassName: Configuration
└─commandClass: 112
└─property: 3
└─endpoint: 0
└─newValue: 0
└─propertyName: Backlight
2023-09-17T17:36:17.060Z - VALUE ADDED
Arg 0:
└─commandClassName: Configuration
└─commandClass: 112
└─property: 4
└─endpoint: 0
└─newValue: 0
└─propertyName: Battery_report
2023-09-17T17:36:17.143Z - VALUE ADDED
Arg 0:
└─commandClassName: Configuration
└─commandClass: 112
└─property: 5
└─endpoint: 0
└─newValue: 5
└─propertyName: Measured_temperature_report
2023-09-17T17:36:17.208Z - VALUE ADDED
Arg 0:
└─commandClassName: Configuration
└─commandClass: 112
└─property: 6
└─endpoint: 0
└─newValue: 0
└─propertyName: Valve_opening_percentage_report
2023-09-17T17:36:17.294Z - VALUE ADDED
Arg 0:
└─commandClassName: Configuration
└─commandClass: 112
└─property: 7
└─endpoint: 0
└─newValue: 1
└─propertyName: window_opening_percentage_report
2023-09-17T17:36:17.375Z - VALUE ADDED
Arg 0:
└─commandClassName: Configuration
└─commandClass: 112
└─property: 8
└─endpoint: 0
└─newValue: 0
└─propertyName: Measured_temperature_offset
2023-09-17T17:36:17.600Z - VALUE ADDED
Arg 0:
└─commandClassName: Notification
└─commandClass: 113
└─property: alarmType
└─endpoint: 0
└─newValue: 0
└─propertyName: alarmType
2023-09-17T17:36:17.603Z - VALUE ADDED
Arg 0:
└─commandClassName: Notification
└─commandClass: 113
└─property: alarmLevel
└─endpoint: 0
└─newValue: 0
└─propertyName: alarmLevel
2023-09-17T17:36:17.654Z - VALUE UPDATED
Arg 0:
└─commandClassName: Notification
└─commandClass: 113
└─property: alarmType
└─endpoint: 0
└─newValue: 0
└─prevValue: 0
└─propertyName: alarmType
2023-09-17T17:36:17.656Z - VALUE UPDATED
Arg 0:
└─commandClassName: Notification
└─commandClass: 113
└─property: alarmLevel
└─endpoint: 0
└─newValue: 0
└─prevValue: 0
└─propertyName: alarmLevel
2023-09-17T17:36:17.658Z - VALUE ADDED
Arg 0:
└─commandClassName: Notification
└─commandClass: 113
└─property: System
└─propertyKey: Hardware status
└─endpoint: 0
└─newValue: 3
└─propertyName: System
└─propertyKeyName: Hardware status
2023-09-17T17:36:17.660Z - INTERVIEW STAGE COMPLETED
Arg 0:
CommandClasses
2023-09-17T17:36:17.663Z - INTERVIEW STAGE COMPLETED
Arg 0:
OverwriteConfig
2023-09-17T17:36:17.665Z - INTERVIEW STAGE COMPLETED
Arg 0:
Complete
2023-09-17T17:36:17.738Z - READY
2023-09-17T17:36:17.758Z - INTERVIEW COMPLETED

```

**Comment by AlCalzone** [comment]
> (for some reason 'add code' isnt formatting the below like it should)

It's a bit counter-intuitive. First paste code, then select it, then press "format as code".

**Comment by RickieCZ** [comment]
Product desc : [https://manual.zwave.eu/backend/make.php?lang=en&sku=EURECOMET&cert=]

**Comment by cadwizzard** [comment]
Thanks to everyone that contributed to adding this ;)

**Comment by Mart124** [comment]
@cadwizzard did you finally manage to tune parameters of your Comet devices ?
We fail [here](https://github.com/home-assistant/core/issues/103254), so would be pleased to have your report.
Thank you 👍

**Comment by cadwizzard** [comment]
Hi.
I gave up I'm afraid, and moved to JSUI. I could alter the parameters...
But I'm really unhappy with the CometZ.

Max reporting detail is 0.5c, unlike the Spirit (I have set to 0.3 normally)

The freeze setpoint is higher than the spirit

Also the spirits actually report detail to 0.01 degree, unlike the comet which is 0.5

Worst of all though is that they seemingly don't report temperature change sometimes reliably. I've had to request a multilevel sensor update to get the right info. This has left my boiler running for hours because the room temperature isn't reported.
I found a re-interview seems to fix that for a while and then they report every 0.5 degree again.

I had 20+ spirits and never had these issues. Upgraded 6 to CometZ. Had the problems on every one. Now can't trust them. Have returned 2 and now am trialling the ZigBee Bosch radiator thermostat 2 in z2mqqt. So far they have been brilliant. I didn't want to move away from z wave, but wanted to modernize the radiators.

I feel like Eurotronic have cheaped out with the hardware on the CometZ compared to the spirits.

Hope you have better luck.

**Comment by bronger** [comment]
I have observed all your issues with this device, plus: If you set a temperature offset, it is not used for the temperature reporting.

**Comment by Mart124** [comment]
Thank you for both your reports.
I plan to use them in Manufacturer specific mode (hence #6583), combined with a room-temperature sensor, and a smart Home Assistant thermostat (for ex. https://github.com/KartoffelToby/better_thermostat / https://github.com/dadge/awesome_thermostat / https://github.com/ScratMan/HASmartThermostat). So fully getting rid of the builtin temperature sensor and logic. Should give better results.

But for now stuck with parameters' configuration :|

**Comment by cadwizzard** [comment]
> But for now stuck with parameters' configuration :|

.... So was I. The only reason I moved to JSUI was to accommodate these devices and the reporting resolution and mirror parameters.

The other 50+ devices were well supported in the native z wave js implementation

But I've found the network map and device management super useful. Especially because of the weirdness I was trying to figure out with the CometZ.

It's such a shame they don't work right.
Also if you change the display to mirrored, the up is setpoint down and down is up as you look at the TRV and it doesn't switch.
The spirits don't have that issue
